lib/script.js

```diff
@@ -40,106 +37,110 @@ Script.prototype.run = function(methodName, globals) {
 };
 
 Script.prototype.createTimeout = function(methodName) {
-  var _this = this;
-
-  if (this.timeoutId){
+  if (this.timeoutId) {
     clearTimeout(this.timeoutId);
     this.timeoutId = null;
   }
 
-  this.timeoutId = setTimeout(function() {
-    _this.timeoutId = null;
-    if (_this.exited) return;
-    _this.exited = true;
-    _this.sandcastle.kickOverSandCastle();
-    _this.emit('timeout', methodName);
+  this.timeoutId = setTimeout(() => {
+    this.timeoutId = null;
+    if (this.exited) { return; }
+    this.exited = true;
+    this.sandcastle.kickOverSandCastle();
+    this.emit('timeout', methodName);
   }, this.timeout);
 };
 
 Script.prototype.reset = function() {
-  if (this.timeoutId){
+  if (this.timeoutId) {
     clearTimeout(this.timeoutId);
     this.timeoutId = null;
   }
   this.exited = false;
 };
 
 Script.prototype.createClient = function(methodName, globals) {
-
-  var _this = this;
-
   try {
-    globals = EJSON.stringify(globals)
-  } catch(err) {
-    _this.onExit(methodName, EJSON.stringify({
+    globals = EJSON.stringify(globals);
+  } catch (err) {
+    this.onExit(methodName, EJSON.stringify({
       error: { message: 'Unserializable Payload' }
     }));
     return;
   }
 
-  this.sandcastle.sandboxReady(function() {
-
-    if (_this.exited) return;
+  this.sandcastle.sandboxReady(() => {
+    if (this.exited) { return; }
 
-    var stream = new BufferStream({ size: 'flexible' });
+    const stream = new BufferStream({ size: 'flexible' });
 
-    var client = net.createConnection(_this.sandcastle.getSocket(), function() {
-      client.write(EJSON.stringify({
-        source: _this.source,// the untrusted JS.
-        sourceAPI: _this.sourceAPI,// the trusted API.
+    const client = net.connect(this.sandcastle.getSocket(), () => {
+      //client.setNoDelay(true);
+      //const sandScript = `sandScript:${process.pid}-${new Date().valueOf()}`;
+      const pack = Buffer.from(EJSON.stringify({
+        source: this.source, // the untrusted JS.
+        sourceAPI: this.sourceAPI, // the trusted API.
         globals: globals, // trusted global variables.
         methodName: methodName
-      }) + '\u0000\u0000'); // the chunk separator
+      }));
+      // timestamp object in case of collision
+      if (this.sandcastle.sharedObject.isClosed()) {
+        this.sandcastle.createMmap();
+      }
+      this.timeStamp = Date.now().toString(32);
```

Comment:
I think I'd like using something like
https://www.npmjs.com/package/nanoid
better, as opposed to timestamp - because while *almost* impossible, I think you *could* get two things at the same millisecond.